src/ble/CHIPBleServiceData.h

```diff
@@ -64,6 +65,20 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint16_t GetAdvertisementVersion() const
+    {
+        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminator) & kAdvertisementVersionMask;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint16_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion &= kAdvertisementVersionMask;
```

Comment:
So to set version to 1, the expectation is to pass in `0x1000`?  This is very much not documented, and a pretty odd API.

Reply:
+1

Reply:
now we can pass 1 to set version to 1